File: code/backend/src/main/kotlin/ro/utcn/sd/icsaszar/project/model/report/TeacherReport.kt
package ro.utcn.sd.icsaszar.project.model.report

import ro.utcn.sd.icsaszar.project.model.activity.Category
import ro.utcn.sd.icsaszar.project.model.activity.Round

class TeacherReport(
        val header: String,
        val rows: List<ReportRow>
)

data class ReportRow(
        val category: Category,
        val subRows: List<ReportSubRow>
)

data class ReportSubRow(
        val round: Round,
        val count: Long
)

File: code/frontend/src/model/objects/report/TeacherReport.ts
import {Round} from "../activity/ActivityEvent";
import {Category} from "../activity/Activity";

export interface TeacherReport {
    header: string;
    rows: ReportRow[];
}

export interface ReportRow {
    category: Category
    subRows: ReportSubRow[]
}

export interface ReportSubRow{
    round: Round
    count: number
}

